server/core.ts

```
@@ -117,21 +138,50 @@ export class WebSocketCore {
         }
-        this.addPlayer(id as string, name as string, ws);
-        this.sendToWebSocket(ws, { type: 'joined', id, name, timestamp: Date.now() });
-        this.broadcastToAll({ 
-          type: 'playerJoin', 
-          data: { id, name, position: { x: 0, y: 0 } }, 
-          timestamp: Date.now() 
-        }, id as string);
+        
+        // Handle position if provided by client
+        const joinPosition = this.validatePosition(restData.position) || { x: 0, y: 0 };
+        
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)